service/service.go

```diff
@@ -147,6 +147,13 @@ type Opts struct {
 	PodmonPort                 string // to indicates the port to be used for exposing podmon API health
 	PodmonPollingFreq          string // indicates the polling frequency to check array connectivity
 	TLSCertDir                 string
+	StorageArrays              map[string]StorageArrayConfig
+}
+
+// StorageArrayConfig represents the configuration of a storage array in the config file
+type StorageArrayConfig struct {
```

Comment:
I know it may seem redundant but adding the arrayID is a good way to double check the structure. Data structures should be independent of the container used to store it. Without the arrayID it cannot stand on its own.